### ISSUE TITLE
`writeln_empty_string` suggests wrong code caused compiler error

### ISSUE THREAD

**Issue by relaxcn** [rust-lang/rust-clippy#16251]
### Summary

When I run clippy on the below code:

```rust
fn issue() {
    let mut v = Vec::new();

    writeln!(v, /* , */ "");
}
```

There is a error:

```rust
warning: empty string literal in `writeln!`
  --> ./test.rs:15:5
   |
15 |     writeln!(v, /* , */ "");
   |     ^^^^^^^^^^^^^^^-------^
   |                    |
   |                    help: remove the empty string
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#writeln_empty_string
```

Fixed code is:

```rust
fn issue() {
    let mut v = Vec::new();

    writeln!(v, /* );
}
```

That causes a compile error.

### Reproducer

Code:

```rust
fn issue() {
    let mut v = Vec::new();

    writeln!(v, /* , */ "");
}
```

Current output:

```rust
warning: empty string literal in `writeln!`
  --> ./test.rs:15:5
   |
15 |     writeln!(v, /* , */ "");
   |     ^^^^^^^^^^^^^^^-------^
   |                    |
   |                    help: remove the empty string
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#writeln_empty_string
```

Desired output:

Correct fix code.

### Version

```text
rustc 1.94.0-nightly (f52090008 2025-12-10)
binary: rustc
commit-hash: f5209000832c9d3bc29c91f4daef4ca9f28dc797
commit-date: 2025-12-10
host: x86_64-pc-windows-msvc
release: 1.94.0-nightly
LLVM version: 21.1.5
```

### Additional Labels

@rustbot label "I-suggestion-causes-error"
